module-bluetooth/Bluetooth/command/Command.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#pragma once
+#include "CommandData.hpp"
+
+namespace bluetooth
+{
+    class Command
+    {
+      public:
+        enum Type
+        {
+            StartScan,
+            StopScan,
+            getDevicesAvailable,
+            VisibilityOn,
+            VisibilityOff,
+            ConnectAudio,
+            DisconnectAudio,
+            PowerOn,
+            PowerOff,
+            Pair,
+            Unpair,
+            StartRinging,
+            StopRinging,
+            StartRouting,
+            StartStream,
+            StopStream,
+            SwitchProfile,
+            CallAnswered,
+            IncomingCallNumber,
+            None,
+        };
+
+        /*
+         *  Calling constructor with data parameter causes creation of internal copy of the data, that
+         *  has to be managed outside the command - this is needed for passing the command's data via
+         *  FreeRTOS queue.
+         */
+        Command(Command::Type type, const std::shared_ptr<CommandData> &data);
+        explicit Command(Command::Type type) : type(type)
+        {}
+        /*
+         * Be aware of calling cleanup() after command with data processing has finished, otherwise
+         * a memory leak will occur.
+         */
+        void cleanup();
```

Comment:
the same goes here @pholat @swichu91

Reply:
Why not change it so that data sent and received is data only? You will be able to create a Command from it - but it it'self would be dumb which would only represent the shared memory sent:
`size_t size, void*/char* data`
Then it would be passed through a parser that would:
```
take data size pair
decode msgpack with:
[command id, size, whatever else]
pass it for further processing :)
```

Reply:
The case is that sometimes we send command without data - when we serialize both command and data to one buffer then indeed, it would make it much simpler, but then during deserializing we would have to somehow process the command type - we loose the polymorphism here.